### PR TITLE
Refactored code and unit tests to account for significantly updated dependencies

### PR DESCRIPTION
**Pull request by mangs**
**Changes:**

- Version bump to `2.0.1`
- Refactored code and unit tests to account for significantly updated dependencies
- Vitest option `--coverage` removed from `test:unit` NPM script and moved into new NPM script `test:unit:coverage` because package `@vitest/coverage-v8` has a bug in it that causes messages to get logged to the console that should not be
- Added ESLint rule to require type imports to be in their own group
- Replaced Vitest coverage reporter `c8` with `v8` as necessitated by the migration to Vitest version `0.32.0`
- `validate:environment` NPM script now executes its subtasks in parallel
- Update GitHub Action workflow files to use Node.js version `18.16.1`
- Update expected internal NPM version to `9.5.1`
- Remove dev dependency `@vscode/ripgrep`, instead rely on locally-installed version because NPM package installs fail too often
- Update dependencies to latest

### CODE REVIEW

File: config/eslint/eslintTypescriptConfig.json [hunk 1]
```diff
@@ -7,8 +7,11 @@
   "extends": [
     "./eslintBaseConfig.json",
     "eslint-config-airbnb-typescript/base",
-    "plugin:@typescript-eslint/recommended",
-    "plugin:@typescript-eslint/recommended-requiring-type-checking",
+    "plugin:@typescript-eslint/recommended-type-checked",
+    "plugin:@typescript-eslint/stylistic-type-checked",
```
Comment (mangs) [inline config/eslint/eslintTypescriptConfig.json:11]: Required for `@typescript-eslint/eslint-plugin` v6+

File: config/vite/vite.config.mts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export default defineConfig({
   test: {
     clearMocks: true,
     coverage: {
-      provider: 'c8',
+      provider: 'v8',
```
Comment (mangs) [inline config/vite/vite.config.mts:24]: `c8` is deprecated and replaced with `v8`

File: tsconfig.json [hunk 3]
```diff
@@ -6,7 +6,6 @@
     "emitDeclarationOnly": true,
     "exactOptionalPropertyTypes": true,
     "forceConsistentCasingInFileNames": true,
-    "importsNotUsedAsValues": "error",
```
Comment (mangs) [inline tsconfig.json:9]: This will be removed from TypeScript soon, so it has to be replaced with rule `verbatimModuleSyntax`

**Review by mangs — COMMENTED** [review]
Some notes for context:

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,17 @@
 # Changelog
 
+## 2.0.1
+
+- Refactored code and unit tests to account for significantly updated dependencies
+- Vitest option `--coverage` removed from `test:unit` NPM script and moved into new NPM script `test:unit:coverage` because package `@vitest/coverage-v8` has a bug in it that causes messages to get logged to the console that should not be
```
Comment (juanma0012) [inline CHANGELOG.md:6]: 👍🏻

**Review by juanma0012 — APPROVED** [review]
LGTM!

**Pull request merged**